File: kafis-practices-backend/Practice.BLL/Models/Admin/AdminReportResponse.cs
using Practice.Application.Models.StudentN;
using Practice.Domain.Core.Common.Enums;
using System;
using System.Collections.Generic;

namespace Practice.Application.Models.Admin
{
    public class AdminReportResponse
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public GradeLevelEnum GradeLevel { get; set; }

        public int OrganizationsAmount { get; set; }
        public IEnumerable<string> OrganizationsNames { get; set; }

        public int AllStudentsAmount { get; set; }
        public int SuccessfulStudentsAmount { get; set; }
        public int FailedStudentsAmount { get; set; }
        public IEnumerable<StudentsGradesSummary> StudentsGradesSummary { get; set; }
    }
}

File: kafis-practices-frontend/libs/interfaces/src/lib/admin-report-response.ts
import { GradeLevel } from "@practice/enums"
import { StudentsGradesSummary } from "./students-grade-summary";

export interface AdminReportResponse {
  year: number;
  gradeLevel: GradeLevel;
  specialty: string;
  organizationsAmount: number;
  organizationsNames: string[];
  allStudentsAmount: number;
  successfulStudentsAmount: number;
  failedStudentsAmount: number;
  studentsGradesSummary: StudentsGradesSummary[];
}
